From 69ae21600a7745979bec0fe8fa661d7916386d4b Mon Sep 17 00:00:00 2001
From: Chuck Short <charles.short@windriver.com>
Date: Mon, 25 Oct 2021 13:23:19 -0400
Subject: [PATCH 1/2] add wheel building support

Add support for building python3 wheel in Debian Openstack.

Add ability to generate python wheels to openstack-pkg-tools. This patch
is needed in order to generate python wheels for openstack clients.

Author: Chuck Short <charles.short@windriver.com>

[refactor for removal of py2 support; save as unified diff]
Signed-off-by: Michel Thebeau <Michel.Thebeau@windriver.com>
---
 build-tools/pkgos-dh_auto_install | 9 +++++++++
 debian/changelog                  | 8 ++++++++
 debian/control                    | 1 +
 3 files changed, 18 insertions(+)

diff --git a/build-tools/pkgos-dh_auto_install b/build-tools/pkgos-dh_auto_install
index f766f13..d663c9e 100755
--- a/build-tools/pkgos-dh_auto_install
+++ b/build-tools/pkgos-dh_auto_install
@@ -18,6 +18,10 @@ for i in $@ ; do
 		PKGOS_IN_TMP=yes
 		shift
 		;;
+	"--wheel")
+		PKGOS_USE_WHEEL=yes
+		shift
+		;;
 	*)
 		;;
 	esac
@@ -36,5 +40,10 @@ fi
 for pyvers in ${PYTHON3S}; do
 	python${pyvers} setup.py install --install-layout=deb --root $(pwd)/debian/${TARGET_DIR}
 done
+if [ "${PKGOS_USE_WHEEL}" = "yes" ]; then
+	for pyvers in ${PYTHON3S}; do
+		python${pyvers} setup.py bdist_wheel --universal  -d $(pwd)/debian/python3-${PY_MODULE_NAME}-wheel/usr/share/python-wheel
+	done
+fi
 rm -rf $(pwd)/debian/python*/usr/lib/python*/dist-packages/*.pth
 rm -rf $(pwd)/debian/tmp/usr/lib/python*/dist-packages/*.pth
diff --git a/debian/changelog b/debian/changelog
index 5630159..027dbef 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,11 @@
+openstack-pkg-tools (135+nmu1) unstable; urgency=medium
+
+  * Non-maintainer upload.
+  * build-tools/pkgos-dh_auto_install: Add wheel support.
+  * rebase Chuck's patch onto 135 (Michel.Thebeau@windriver.com)
+
+ -- Chuck Short <zulcss@ubuntu.com>  Sun, 03 Oct 2021 15:10:16 +0000
+
 openstack-pkg-tools (135) unstable; urgency=medium
 
   * Add a new $CUSTOM_SYSTEMD_SERVICE_DIRECTIVES variable.
diff --git a/debian/control b/debian/control
index 8cb1b2d..70ed450 100644
--- a/debian/control
+++ b/debian/control
@@ -17,6 +17,7 @@ Architecture: all
 Multi-Arch: foreign
 Depends:
  python3-pip,
+ python3-wheel,
  gettext,
  jq,
  po-debconf,
-- 
2.34.1

